### ISSUE TITLE
Confusing (or misleading) yellow popup message after copy/paste of tags in the editor

### ISSUE THREAD

**Issue by jajajaneeneenee** [MarcusWolschon/osmeditor4android#1511]
## Vespucci Version
16.1.1

## Download source
Google Play Store

## Behaviour/Symptoms
If I copy some tags from an object (in the editor, detail view) and paste it in another object (in the editor), a yellow pop up message appears with the text “Note - preset overwrote tag(s)” – and it seems to be new (?), at least I didn't noticed it in the past:

![Screenshot_50%](https://user-images.githubusercontent.com/48966999/145662229-c38c5186-0e1c-4d26-b551-2c5c4c996cd2.png)

It confused me a little, because tags have not been overwritten by selecting a preset, but by pasting tags (and it was just intended that existing values SHOULD BE overwritten – if they exist at all – or simply set if empty or not existing before). I was wondering if something went wrong with the preset or indirect by a new preset matching when I saw the message ... (Although I didn't copied/pasted tags that changed the matching of the preset – that's why it was even more confusing). I linked the message text directly to changing of tags by assigning a preset with other fixed key/value pairs than the current object – and that something could have been changed what wasn't intended. Then you have to look to EVERY tag if you want to be sure – and it can be very difficult if an object (e.g. a road) has lots of tags.

Perhaps this message text is wrong or not precise enough (or only misleading) here? For me it would be clearer if it would be someting like: “Note – the pasting has overwritten one or more existing tags“. Or simply: “Note – one or more existing tags have been overwritten“.

Additionally (or even worse), the message also appears if NO values have been overwritten, but onIy set – I mean if the pasted keys (with values) didn't have any values in the target object yet (keys are there, because part of the preset item, but with empty values) – that's also confusing. I would prefer if it would only appear if some (existing) values REALLY have been overwritten (= replaced). Or the message should be: “Note – one or more tags/keys/values have been set/added.“ But that's exactly what you expect if you paste something what you copied consciously before from another object – perhaps a pop up message is not absolutly neccessary in such a case? On the other hand it could be a helpful warning if you accidentally touched the paste button with something unwanted in the clipboard ... but then a precise text would be helpful – perhaps even with the number of tags which have been modified/set? It would be a confirmatory message about what you did (intentionally), and make you feel good (or warn you in a good way if something went wrong). 

E.g.: “Note – 5 tags were overwritten (or set).“ / “Note – 1 tag was overwritten (or set).“

Or even: “Note – 2 tags were overwritten and 3 were added.“ / “Note – 1 tag was overwritten.“ / “Note – 1 tag was added.“ / ...  I know ... some cases need to be considered, but I would appriciate it ... If you want it less user friendly, but more coding friendly: “Note – tags were overwritten (X tags) and/or added (Y tags).“ with X and Y for the number of cases.

Note: It seems that the message ONLY DOESN'T appear after pasting if you paste tags which are not part of the current preset (and if these tags didn't exist in the target object – or they exist with an empty value). Or if you paste something without any difference in the currect object (both is OK). But I am not 100% sure ... I haven't checked every combination.

Perhaps you could take a look to the whole procedure and make it more ”watertight“ (regarding the clearness of the displayed information)? Using copy/paste of single tags to modify objects and tags is quite essential, I think. It should be very clear what is happening.

**Comment by simonpoole** [comment]
The message text was simply wrong because the (existing) correct message wasn't being used.

However I can't recreate the issue with the message appearing when empty or non-existing keys are being set, the code explicitly ignores these, could you give a concrete example of this.

**Comment by simonpoole** [comment]
@jajajaneeneenee ping

**Comment by jajajaneeneenee** [comment]
Hello .... sorry ... I'm currently not the fastest with answering ...

I can reproduce that a yellow message (now with the text "Note – operation overwrote tags") appears when pasting a key/value and the corresponding key is **empty** with with these steps:

1. Selected a highway=residential and set e.g. incline=10% (wasn't set before) any copy this tag (key+value) in the "Details" tab of the editor
2. Select another highway=* street (which has no incline key/value set)
3. Press the "+ abc" button first => incline is listed now in the editor (with an empty value)
4. Press the paste button

Notes: 
* If I press the undo button directly after step 4 and after that the paste button again, there is **no message**. (But then, only the non-optional keys are shown again. If repeating **step 3 and 4**, the message comes again ...)
* If I paste it to an object which hasn't the key "incline" in its preset (e.g. barrier=fence) – just for a test –, there is no message. 

And you are not planning to implement the luxury message “Note – tags were overwritten (X tags) and/or added (Y tags).“? Just to ask ...

**Comment by simonpoole** [comment]
> And you are not planning to implement the luxury message “Note – tags were overwritten (X tags) and/or added (Y tags).“? Just to ask ...

As a general rule I try to restore the intended behaviour and then add enhancements whenever possible. As this needs new output (with all related stuff like translations and so on), it might make more sense to actually list the overwritten tags or similar.
